/* -------------------------------------------------------------------------- */
/* Copyright 2002-2025, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */
syntax = "proto3";

import "shared.proto";

package one.hook;

option go_package = "github.com/OpenNebula/one/src/oca/go/src/goca/api/hook";

message AllocateRequest
{
  string session_id = 1;
  string template   = 2;
}

message DeleteRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message InfoRequest
{
  string session_id = 1;
  int32 oid         = 2;
  bool decrypt      = 3;
}

message UpdateRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string template   = 3;
  int32 append      = 4;
}

message RenameRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string name       = 3;
}

message LockRequest
{
  string session_id = 1;
  int32 oid         = 2;
  int32 level       = 3;
  bool test         = 4;
}

message UnlockRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message RetryRequest
{
  string session_id = 1;
  int32 oid         = 2;
  int32 hk_exe_id   = 3;
}

message PoolInfoRequest
{
  string session_id = 1;
}

message LogInfoRequest
{
  string session_id = 1;
  int32 min_ts      = 2;
  int32 max_ts      = 3;
  int32 hook_id     = 4;
  int32 rc_hook     = 5;
}

service HookService
{
  rpc Allocate (one.hook.AllocateRequest) returns (one.ResponseID);

  rpc Delete (one.hook.DeleteRequest) returns (one.ResponseID);

  rpc Update (one.hook.UpdateRequest) returns (one.ResponseID);

  rpc Rename (one.hook.RenameRequest) returns (one.ResponseID);

  rpc Lock (one.hook.LockRequest) returns (one.ResponseID);

  rpc Unlock (one.hook.UnlockRequest) returns (one.ResponseID);

  rpc Retry (one.hook.RetryRequest) returns (one.ResponseID);

  rpc Info (one.hook.InfoRequest) returns (one.ResponseXML);

  rpc PoolInfo (one.hook.PoolInfoRequest) returns (one.ResponseXML);

  rpc LogInfo (one.hook.LogInfoRequest) returns (one.ResponseXML);
}
